From 6914a7e96b73fb9c49640856d94658c78382ed4f Mon Sep 17 00:00:00 2001
From: Guy Nisan <gnisan@habana.ai>
Date: Tue, 15 Jun 2021 23:17:47 +0300
Subject: [PATCH] [SW-45184] i2c: fix nvme mi compatibility

Fix NVMe MI compatibility by returning the code length in offset 0. Keep
old logic for Goya.

Change-Id: Ibcf22266f1da5e86263065dadd0e2eeca09bd1bb
Signed-off-by: Guy Nisan <gnisan@habana.ai>
---

diff --git a/arch/goya/common.c b/arch/goya/common.c
index 6488f2d..d47d8e2 100644
--- a/arch/goya/common.c
+++ b/arch/goya/common.c
@@ -261,3 +261,8 @@
 }
 
 void spi_prot_early(void) {}
+
+bool is_nvme_mi_compatible(void)
+{
+	return false;
+}
diff --git a/preboot/module_i2c_slave.c b/preboot/module_i2c_slave.c
index 68a2797..ef1f03c 100644
--- a/preboot/module_i2c_slave.c
+++ b/preboot/module_i2c_slave.c
@@ -53,12 +53,22 @@
 	}
 }
 
+__weak bool is_nvme_mi_compatible(void)
+{
+	return true;
+}
+
 u8 i2cslave_handle_read_req(u8 cmd, u32 read_seq)
 {
 	u8 ret;
 
 	switch (cmd) {
 	case HL_ASIC_TEMP_CMD_CODE:
+		if (is_nvme_mi_compatible())
+			ret = HL_ASIC_TEMP_CMD_CODE_LEN;
+		else
+			ret = get_composite_temp();
+		break;
 	case HL_ASIC_TEMP_MAX_VAL:
 		ret = get_composite_temp();
 		break;
